scripts/base_network.py

```diff
@@ -281,30 +272,60 @@ def _load_transformers_from_osm(fp_osm_transformers, buses):
     return transformers
 
 
-def _set_electrical_parameters_lines(lines_config, voltages_config, lines):
+def _get_linetypes(line_types, voltages):
+    """
+    Return the linetypes for selected voltages.
+    """
+    vnoms_diff = set(voltages).symmetric_difference(set(line_types.keys()))
+    if vnoms_diff:
+        logger.warning(
+            f"Voltages {vnoms_diff} not in the {line_types} or {voltages} list."
+        )
+    return {k: v for k, v in line_types.items() if k in voltages}
+
+
+def _get_linetype_by_voltage(v_nom, d_linetypes):
+    """
+    Return the linetype based on the voltage.
+    """
+    v_nom_min, line_type_min = min(
+        d_linetypes.items(),
+        key=lambda x: abs(x[0] - v_nom),
+    )
+    return line_type_min
```

Comment:
Yeah; I think it is ok.
A bare conducture can be used in a wide range of voltages and its insulation is guaranteed by insulators rather than the conductor itself.
The ability of a conductor to be used at different voltages is indirect: higher voltages imply higher heights, thus higher wind speeds, thus higher mechanical stresses for example.

I prefer this approach as for non-standard voltages, we may miss the allocation thus leading to issues.
Moreover, even if "better standard values" could be used, the error may be small and acceptable.
I'd personally like to extend the standard values to lower values to better reflect global values

Reply:
Agree, the incorrect allocation of the line type seems to be the major issue for now.

Thanks a lot for the detailed explanation!